extensions/clusters/clusters.go

```diff
@@ -450,19 +455,26 @@ func NewK3SRKE2ClusterConfig(clusterName, namespace string, clustersConfig *Clus
 		MachinePools:         machinePools,
 	}
 
+	agentEnvVars := []rkev1.EnvVar{}
```

Comment:
can't this be omitted, or does setting an initial object match more closely to what rancher does to create a cluster? (thinking of the case where no env vars are set, this was nil before this PR)

Reply:
Good question.